src/Integration/Binding/BindingWorkflow.cs

```diff
@@ -320,10 +339,84 @@ private void FinishBindingOnUIThread(IProgressController controller, IProgressSt
             }
         }
 
+        internal /*for testing purposes*/ void UnpackAdditionalFiles(IProgressController controller, CancellationToken token, IProgressStepExecutionEvents notificationEvents)
+        {
+            if (!this.AdditionalFiles.Any())
+            {
+                return; // Nothing to unpack
+            }
+
+            Solution2 solution = this.projectSystemHelper.GetCurrentActiveSolution();
+            if (solution == null)
+            {
+                Debug.Fail("Cannot get current active solution");
+                return;
+            }
+
+            string solutionRoot = Path.GetDirectoryName(solution.FullName);
+            string root = this.solutionRuleSetWriter.GetOrCreateRuleSetDirectory(solutionRoot);
+
+            DeterminateStepProgressNotifier notifier = new DeterminateStepProgressNotifier(notificationEvents, this.AdditionalFiles.Count());
+
+            foreach (var additionalFile in this.AdditionalFiles)
+            {
+                if (token.IsCancellationRequested)
+                {
+                    break;
+                }
+
+                string message = string.Format(CultureInfo.CurrentCulture, Strings.UnpackingAdditionalFile, additionalFile.FileName);
+                notifier.NotifyIncrementedProgress(message);
+
+                string filePath = Path.Combine(root, additionalFile.FileName);
+                File.WriteAllBytes(filePath, additionalFile.Content);
+
+                this.AdditionalFilePaths.Add(filePath);
+            }
+        }
+
+        internal /*for testing purposes*/ void InjectAdditionalFiles()
+        {
+            // Add to solution items
+            Project solutionItemsProject = this.projectSystemHelper.GetSolutionItemsProject();
+            if (solutionItemsProject == null)
+            {
+                Debug.Fail("Could not find the solution items project");
+            }
+            else
+            {
+                foreach (var additionalFilePath in this.AdditionalFilePaths)
+                {
+                    this.projectSystemHelper.AddFileToProject(solutionItemsProject, additionalFilePath);
+                }
+            }
+
+            // Inject into projects
+            foreach (var project in this.projectSystemHelper.GetSolutionManagedProjects())
+            {
+                string projectRoot = PathHelper.ForceDirectoryEnding(Path.GetDirectoryName(project.FullName));
```

Comment:
Shouldn't CalculateRelativePath force directory ending instead of sprinkling this in the calling code?